trojanscheduler/views.py

```diff
@@ -0,0 +1,25 @@
+# I think these work in principle, but I don't know how to loop through all the possible objects
+# this only works if you call it on a specific object
+
+from models import BasicClassInfo, BasicClassSection
+
+
+class BasicClassInfoView:
+    # Create a view that will return a list of basicclasssection objects based on class_code and class_section
+    # Ex: If the query passes in "CSCI 102" expect to see all sections that contains CSCI 102
+    #    If the query passes in "all" return all classes in basicclassinfo objects ordered alphabetically
+
+    def class_section_search(object, query):
+        if query == "all":
+            return object
+        code = query[: query.index(" ")]
+        section = query[query.index(" ") + 1 :]
+        if object.class_code == code and object.class_section == section:
+            return object
+
+    # Create another view that returns a list of basicclassinfo objects based on class code
+    # If the query passes in is "CSCI" expect to see all basicclassinfo that contains "CSCI" so it'll have CSCI 102, CSCI 103, ...
+
+    def class_code_search(object, code):
+        if object.class_code.equals(code):
+            return object.class_code + " " + object.class_section
```

Comment:
Sorry for the late feedback. Refer to https://github.com/markkyaw/trojan_scheduler/blob/main/api/v1/classes/views/views.py to get an idea on how `view` files should be formatted.
Basically, you'll need to create corresponding `urls.py` as shown in https://github.com/markkyaw/trojan_scheduler/blob/main/api/v1/classes/urls.py
Create a new `url` and call it like `class_search` or something. And the view file will look like the class based view listed here
https://www.django-rest-framework.org/tutorial/3-class-based-views/
Let me know if you wanna pair program on this :)

Reply:
To make it search, you'll be doing
`BasicClassSection.objects.filter`
You can read more about them here
https://docs.djangoproject.com/en/4.0/topics/db/queries/